ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -180,7 +180,9 @@ private void onChunkOperationComplete(GetChunk chunk) {
           // GetManager remove this operation.
           operationCompleted = true;
           List<StoreKey> chunkIds = e == null && compositeBlobInfo != null ? compositeBlobInfo.getKeys() : null;
-          operationResult = new GetBlobResultInternal(null, chunkIds);
+          // blobInfo.setStoreKeys(chunkIds); // this is not good. storeKeys don't belong blobInfo but getCallback only accept blobInfo
+          // operationResult = new GetBlobResultInternal(null, chunkIds);
+          operationResult = new GetBlobResultInternal(new GetBlobResult(blobInfo, blobDataChannel, chunkIds), chunkIds); // add blob info here
```

Comment:
we might not need GetBlobResultInternal anymore if GetBlobResult already contains the chunk IDs.

Reply:
Yes. Get rid of GetBlobResultInternal make code cleaner, but no sure if we have any particular reason to use GetBlobResultInternal. @vgkholla @pnarayanan